lib/service/service.go

```diff
@@ -179,24 +200,48 @@ func (process *TeleportProcess) connectToAuthService(role teleport.Role) (*Conne
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	storage := utils.NewFileAddrStorage(
-		filepath.Join(process.Config.DataDir, "authservers.json"))
-
-	authUser := identity.Cert.ValidPrincipals[0]
-	authClient, err := auth.NewTunClient(
-		string(role),
-		process.Config.AuthServers,
-		authUser,
-		[]ssh.AuthMethod{ssh.PublicKeys(identity.KeySigner)},
-		auth.TunClientStorage(storage),
-	)
-	// success?
+	tlsConfig, err := identity.TLSConfig()
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err)
+		}
+		// connect using legacy SSH and get new set of TLS credentials
+		storage := utils.NewFileAddrStorage(
+			filepath.Join(process.Config.DataDir, "authservers.json"))
+
+		authUser := identity.Cert.ValidPrincipals[0]
```

Comment:
Is the `ValidPrincipals` length checked anywhere?

Reply:
yes auth/init.go:696